### DISCUSSION TITLE
What is the best way to modify an existing binding?

### DISCUSSION THREAD

**Discussion by jspv** [Textualize/textual#1562] [Q&A] [answered]
In the pre-CSS textual, bind was a method which could be called to change an existing binding.  What is the equivalent in the new BINDINGS model?  

My goal:  My applications have different states and I would like re-bind keys to take different actions depending on that state and clearly note in the Footer what action will be taken.   For example, one of my bindings toggles a power relay - I'd like to clearly show in the Footer that a key will "Enable Relay" or "Disable Relay".  Today it just says "Toggle Power Relay" and  my action checks the state to make the appropriate change, but I'd like to be more descriptive by changing the binding.

**Comment by jspv** [comment, marked as answer]
Answering my own question, the bind method is still there and works just fine.  I somehow had convinced myself that the new BINDINGS model had caused bind to be deprecated.

**Reply by fohrloop** [reply]
Is there a public interface for this? I found `self._bindings` on a `DataTable`  (subclass of `Widget`) which is a `BindingsMap` object with `bind()` and `key_to_bindings` which both seem really useful.
